substrate/frame/revive/src/vm/evm/instructions/host.rs

```diff
@@ -199,56 +181,75 @@ pub fn sstore<'ext, E: Ext>(context: Context<'_, 'ext, E>) {
 
 	popn!([index, value], context.interpreter);
 
-	let Some(state_load) =
-		context.host.sstore(context.interpreter.input.target_address(), index, value)
-	else {
+	let key = Key::Fix(index.to_be_bytes());
+	let take_old = false;
+	let Ok(write_outcome) = context.interpreter.extend.set_storage(
+		&key,
+		Some(value.to_be_bytes::<32>().to_vec()),
+		take_old,
+	) else {
 		context.interpreter.halt(InstructionResult::FatalExternalError);
 		return;
 	};
-
-	// EIP-1706 Disable SSTORE with gasleft lower than call stipend
-	if context.interpreter.runtime_flag.spec_id().is_enabled_in(ISTANBUL) &&
-		context.interpreter.gas.remaining() <= CALL_STIPEND
-	{
-		context.interpreter.halt(InstructionResult::ReentrancySentryOOG);
-		return;
-	}
-	gas_legacy!(
+	gas!(
 		context.interpreter,
-		gas::sstore_cost(
-			context.interpreter.runtime_flag.spec_id(),
-			&state_load.data,
-			state_load.is_cold
-		)
+		RuntimeCosts::SetStorage { old_bytes: write_outcome.old_len(), new_bytes: 32 }
 	);
-
-	context.interpreter.gas.record_refund(gas::sstore_refund(
```

Comment:
should we call `adjust_gas` ?

Reply:
Please see the implementation for PVM:
```
	fn set_storage(
		&mut self,
		memory: &M,
		flags: u32,
		key_ptr: u32,
		key_len: u32,
		value: StorageValue,
	) -> Result<u32, TrapReason>
```
first you charge the max and then you refund.